### PR TITLE
ROX-22818: Move automation-flavor image version to a Chart annotation

### PR DESCRIPTION
**Pull request by tommartensen**
This will make it easier to update the image versions consistently. 

Note for reviewers:

- Some workflows have fields with `"{{workflow.parameters...}}"`. The double quotes are necessary to avoid interpretation of the `{{` as objects and subsequent YAML parsing errors. I switched to single quotes `'`, because helm expects double quotes for its templating engine, `"{{...}}"` becomes `'{{ "{{" }}...{{ "}}"}}'`. Where unnecessary, I removed doubles quotes, e.g. `"--argument={{ ... }}"` does not need the double quotes, because it is interpreted as a string. enjoy the review :'( 
- All flavors were smoke tested in the latest version, see #1208 for details. There is https://github.com/stackrox/automation-flavors/pull/216 pending with two fixes.

### CODE REVIEW

**Review by gavin-stackrox — CHANGES_REQUESTED** [review]
Lets discuss in Slack. Having all infra flavors using the same automation-flavor presents some challenges.

**Comment by tommartensen** [comment]
Agreement in team sync: @tommartensen does a manual smoke test for all flavors before merging this PR.

File: .github/workflows/smoke-test.yaml [hunk 1]
```diff
@@ -27,3 +25,48 @@ jobs:
...
+        env:
+          INFRA_TOKEN: ${{ env.INFRA_TOKEN_PROD }}
+        run: |
+          set -uo pipefail
+          infractl artifacts "${{ inputs.cluster }}" -d "/home/runner/artifacts" >> "$GITHUB_STEP_SUMMARY"
+          kubectl -n infra port-forward svc/infra-server-service 8443:8443 &
+          sleep 10
+
+      - name: Create infra cluster for ${{ matrix.name }}
+        uses: stackrox/actions/infra/create-cluster@tm/create-cluster-endpoint
```
Comment (tommartensen) [inline .github/workflows/smoke-test.yaml:64]: Change to `v1` when https://github.com/stackrox/actions/pull/48 is merged

**Review by gavin-stackrox — APPROVED** [review]
LGTM! `static` should be renamed now right?

**Comment by tommartensen** [comment]
> static should be renamed now right?

eventually maybe.

**Pull request merged**